optimum/neuron/generation/utils.py

```diff
@@ -1080,7 +1080,9 @@ def generate(
             )
 
         else:
-            # 10.a go into not optimized generation modes
+            # 10.a go into not optimized generation modes. We can't handle KV cache in this case.
+            assert model_kwargs['use_cache'] == False, "`use_cache` is currently not supported for this sampling method."
```

Comment:
Can we reword that as an exception please?
We do not do asserts in the main code in `optimum-neuron`.

Reply:
Yes, changed it to raising ValueError intead.